src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleKeyCache.scala

```diff
@@ -60,27 +59,27 @@ class GoogleKeyCache(val googleIamDAO: GoogleIamDAO, val googleStorageDAO: Googl
   }
 
   private def keyNamePrefix(project: GoogleProject, saEmail: WorkbenchEmail) = s"${project.value}/${saEmail.value}"
-  private def keyNameFull(project: GoogleProject, saEmail: WorkbenchEmail, keyId: ServiceAccountKeyId) = s"${keyNamePrefix(project, saEmail)}/${keyId.value}"
+  private def keyNameFull(project: GoogleProject, saEmail: WorkbenchEmail, keyId: ServiceAccountKeyId) = GcsObjectName(s"${keyNamePrefix(project, saEmail)}/${keyId.value}")
 
   private def furnishNewKey(pet: PetServiceAccount): Future[String] = {
     val keyFuture = for {
       key <- OptionT.liftF(googleIamDAO.createServiceAccountKey(pet.id.project, pet.serviceAccount.email) recover {
         case e: GoogleJsonResponseException if e.getDetails.getCode == StatusCodes.TooManyRequests.intValue => throw new WorkbenchException("You have reached the 10 key limit on service accounts. Please remove one to create another.")
       })
       decodedKey <- OptionT.fromOption[Future](key.privateKeyData.decode)
-      _ <- OptionT.liftF(googleStorageDAO.storeObject(googleServicesConfig.googleKeyCacheConfig.bucketName, keyNameFull(pet.id.project, pet.serviceAccount.email, key.id), new ByteArrayInputStream(decodedKey.getBytes)))
+      _ <- OptionT.liftF(googleStorageDAO.storeObject(googleServicesConfig.googleKeyCacheConfig.bucketName, keyNameFull(pet.id.project, pet.serviceAccount.email, key.id), new ByteArrayInputStream(decodedKey.getBytes), "text/plain"))
     } yield decodedKey
 
     keyFuture.value.map(_.getOrElse(throw new WorkbenchException("Unable to furnish new key")))
   }
 
-  private def retrieveActiveKey(pet: PetServiceAccount, keyObjects: List[StorageObject]): Future[String] = {
-    val mostRecentKey = keyObjects.sortBy(_.getTimeCreated.getValue).last
-    val keyRetired = System.currentTimeMillis() - mostRecentKey.getTimeCreated.getValue > 86400000 * googleServicesConfig.googleKeyCacheConfig.activeKeyMaxAge
+  private def retrieveActiveKey(pet: PetServiceAccount, keyObjects: List[GcsObjectName]): Future[String] = {
+    val mostRecentKey = keyObjects.sortBy(_.timeCreated.toEpochMilli).last
+    val keyRetired = System.currentTimeMillis() - mostRecentKey.timeCreated.toEpochMilli > 86400000 * googleServicesConfig.googleKeyCacheConfig.activeKeyMaxAge
```

Comment:
note to self & reviewers: this causes a Long overflow in unit tests. i'll look into how i broke this shortly